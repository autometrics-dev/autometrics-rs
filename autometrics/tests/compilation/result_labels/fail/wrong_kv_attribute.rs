// This test ensures that the macro fails with a readable
// error when the attribute given to one variant inside the
// enumeration is not in the correct form.
use autometrics_macros::ResultLabels;

struct Inner {}

#[derive(ResultLabels)]
enum MyError {
    Empty,
    #[label = "error"]
    ClientError {
        inner: Inner,
    },
    ServerError(u64),
}

fn main() {}
