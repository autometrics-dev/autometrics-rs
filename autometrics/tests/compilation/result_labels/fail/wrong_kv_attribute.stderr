error: Only `label(result = "RES")` (RES can be "ok" or "error") is supported
  --> tests/compilation/result_labels/fail/wrong_kv_attribute.rs:11:7
   |
11 |     #[label = "error"]
   |       ^^^^^^^^^^^^^^^
